jetty-core/jetty-quic/jetty-quic-common/src/main/java/org/eclipse/jetty/quic/common/QuicSession.java

```diff
@@ -401,8 +401,17 @@ public void outwardClose(long error, String reason)
         if (LOG.isDebugEnabled())
             LOG.debug("outward closing 0x{}/{} on {}", Long.toHexString(error), reason, this);
         quicheConnection.close(error, reason);
-        // Flushing will eventually forward the outward close to the connection.
-        flush();
+        try
+        {
+            // Flushing will eventually forward the outward close to the connection.
+            flush();
+        }
+        catch (IllegalStateException ise)
+        {
+            // Flusher already is in CLOSED state, nothing else to do.
+            if (LOG.isDebugEnabled())
+                LOG.debug("IllegalStateException caught while flushing, flusher={} {}", flusher, this, ise);
+        }
```

Comment:
`flush()` is called from many places, I think it would be better to try/catch inside it, rather than only here.

Reply:
In my view, the only time it is safe to catch the iterating callback's ISE is when we know we're closing anyway. Other callers of `flush()` should get the exception as that most likely indicates a bug, shouldn't they?